torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -402,6 +429,58 @@ std::shared_ptr<FutureMessage> TensorPipeAgent::send(
   return futureResponseMessage;
 }
 
+void TensorPipeAgent::pollTimeoutRpcs() {
+  while (rpcAgentRunning_.load()) {
+    std::unique_lock<std::mutex> lock(timeoutMapMutex_);
+
+    steady_clock_time_point earliestTimeout =
+        std::chrono::steady_clock::now() + kLargeTimeDuration;
+
+    // We sleep until the earliest expiring RPC in the timeoutMap_. We must
+    // also ensure that we sleep while the map is empty, and we exit sleeping
+    // if the RPC Agent has been shutdown.
+    for (;;) {
+      if (!rpcAgentRunning_.load()) {
+        return;
+      }
+      if (std::chrono::steady_clock::now() >= earliestTimeout) {
+        break;
+      }
+      if (!timeoutMap_.empty()) {
```

Comment:
```suggestion
      steady_clock_time_point earliestTimeout =
          std::chrono::steady_clock::now() + kLargeTimeDuration;
      if (!timeoutMap_.empty()) {
```
You could consider moving the definition of `earliestTimeout` inside the loop, as it's not used outside of it.